core/kazoo_attachments/src/gen_attachment.erl

```diff
@@ -0,0 +1,84 @@
+-module(gen_attachment).
+
+%% Setter
+-export([error_response/2]).
+%% Getters
+-export([error_code/1, error_body/1]).
+%% Helper(s)
+-export([is_error_response/1]).
+
+-include("kz_att.hrl").
+
+-type settings()         :: kz_data:connection().
+-type db_name()          :: kz_term:ne_binary().
+-type doc_id()           :: kz_term:ne_binary().
+-type att_name()         :: kz_term:ne_binary().
+-type contents()         :: kz_term:ne_binary().
+-type options()          :: kz_data:options().
+-type handler_props()    :: kz_data:connection().
```

Comment:
i don't see the need for creating these types, just use the ones already defined

Reply:
We had a talk at one of our stand up sessions and we agreed to let it be.